packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmAnyNestedCollectionTests.kt

```diff
@@ -0,0 +1,788 @@
+import io.realm.kotlin.ext.realmSetOf
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.test.common.utils.assertFailsWithMessage
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.types.RealmAny
+import org.mongodb.kbson.ObjectId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class RealmAnyNestedCollectionTests {
+
+    private lateinit var configBuilder: RealmConfiguration.Builder
+    private lateinit var configuration: RealmConfiguration
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        configBuilder = RealmConfiguration.Builder(
+            setOf(
+                JsonStyleRealmObject::class,
+                Sample::class,
+            )
+        ).directory(tmpDir)
+        configuration = configBuilder.build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun setInRealmAny_copyToRealm() = runBlocking {
+        val sample = Sample().apply { stringField = "SAMPLE" }
+        realm.write {
+            val instance = JsonStyleRealmObject().apply {
+                value = RealmAny.create(
+                    realmSetOf(
+                        RealmAny.create(5),
+                        RealmAny.create("Realm"),
+                        RealmAny.create(sample),
+                    )
+                )
+            }
+            copyToRealm(instance)
+        }
+        val anyValue: RealmAny = realm.query<JsonStyleRealmObject>().find().single().value!!
+        assertEquals(RealmAny.Type.SET, anyValue.type)
+        anyValue.asSet().toMutableSet().let { embeddedSet ->
+            assertTrue { embeddedSet.remove(RealmAny.create(5)) }
+            assertTrue { embeddedSet.remove(RealmAny.create("Realm")) }
+            assertEquals("SAMPLE", embeddedSet.single()!!.asRealmObject<Sample>().stringField)
+        }
+    }
+
+    @Test
+    fun setInRealmAny_assignment() = runBlocking {
+        val sample = Sample().apply { stringField = "SAMPLE" }
+        realm.write {
+            val instance = copyToRealm(JsonStyleRealmObject())
+            instance.value = RealmAny.create(
+                realmSetOf(
+                    RealmAny.create(5),
+                    RealmAny.create("Realm"),
+                    RealmAny.create(sample),
+                )
+            )
+            instance
+        }
+        val anyValue: RealmAny = realm.query<JsonStyleRealmObject>().find().single().value!!
+        assertEquals(RealmAny.Type.SET, anyValue.type)
+        anyValue.asSet().toMutableSet().let { embeddedSet ->
```

Comment:
It was just to verify the elements of the set by removing them one by one from a mutable copy. I have reworked the test to just verify the elements with `size` and `contains` to avoid the confusion with `toMutableSet()` here.